src/backend/app/tasks/task_routes.py

```diff
@@ -53,9 +54,8 @@ async def new_event(
 
     match detail.event:
         case EventType.REQUESTS:
-            # TODO: Combine the logic of `update_or_create_task_state` and `request_mapping` functions into a single function if possible. Will do later.
             project = await get_project_by_id(db, project_id)
```

Comment:
Could we possibly get the project from `project.deps.get_project_by_id`?
Using `Depends`?

The project_crud.get_project_by_id function can probably be deleted

Reply:
Yes, we can indeed get the project using project.deps.get_project_by_id with Depends. I’ve now removed the project_crud.get_project_by_id function as it is redundant. Thanks for pointing that out!